libcudacxx/include/cuda/std/__cccl/grid_constant.h

```diff
@@ -0,0 +1,49 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __CCCL_GRID_CONSTANT_H
+#define __CCCL_GRID_CONSTANT_H
+
+#include <cuda/std/__cccl/compiler.h>
+#include <cuda/std/__cccl/system_header.h>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+/// In device code, _CCCL_PTX_ARCH expands to the PTX version for which we are compiling.
+/// In host code, _CCCL_PTX_ARCH's value is implementation defined.
+#ifndef _CCCL_PTX_ARCH
+#  if defined(_CCCL_COMPILER_NVHPC)
+// __NVCOMPILER_CUDA_ARCH__ is the target PTX version, and is defined when compiling both host code and device code.
+// Currently, only one PTX version can be targeted.
+#    define _CCCL_PTX_ARCH __NVCOMPILER_CUDA_ARCH__
```

Comment:
Don't use `__NVCOMPILER_CUDA_ARCH__`.  That macro was intended to be temporary hack and has been obsolete for several years.  NVC++ can compile for multiple GPU architectures at the same time, so defining a single value for `_CCCL_PTX_ARCH` doesn't make a lot of sense.  The best that can be done is to use `__CUDA_MINIMUM_ARCH__`, which is defined in `<nv/target>` for both NVC++ and NVCC.  For NVC++ it is the oldest GPU architecture in the current compilation.  (This comment may be moot, because Michael has concerns about this entire section.)

Reply:
thanks. Unfortunately, I still see usage of `__NVCOMPILER_CUDA_ARCH__` in CUB and Thrust